src/visGeometry/index.ts

```diff
@@ -1116,18 +1117,17 @@ class VisGeometry {
         this.setAgentColors();
     }
 
-    public applyColorToAgents(
-        agentIds: number[],
-        color: string | number
-    ): void {
-        const newColorData = this.colorHandler.setColorForAgentTypes(
-            agentIds,
-            color
-        );
-        this.renderer.updateColors(
-            newColorData.numberOfColors,
-            newColorData.colorArray
-        );
+    public applyColorToAgents(colorAssignments: ColorAssignment[]): void {
+        colorAssignments.forEach((color) => {
+            const newColorData = this.colorHandler.setColorForAgentTypes(
+                color.agentIds,
+                color.color
+            );
+            this.renderer.updateColors(
+                newColorData.numberOfColors,
+                newColorData.colorArray
+            );
+        });
```

Comment:
@meganrm pointing out the most significant change in response to @frasercl feedback since your approval.

Changed `applyColorToAgents` to receive array of arguments and back changes so there is only one call to `updateScene` when applying color changes.